### ISSUE TITLE
False positives in spaces_left_parentheses_linter

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#780]
All of the following generate lints currently; I don't think any of them should:

```
lintr::lint("x**(y + z)\n", lintr::spaces_left_parentheses_linter())
lintr::lint("'[[<-.data.frame'(object, y)\n", lintr::spaces_left_parentheses_linter())
lintr::lint("object@data@get('input')\n", lintr::spaces_left_parentheses_linter())
lintr::lint("x <- ~(. +y)\n", lintr::spaces_left_parentheses_linter())
lintr::lint("for (ii in 1:10) next()\n", lintr::spaces_left_parentheses_linter())
lintr::lint("for (ii in 1:10) break()\n", lintr::spaces_left_parentheses_linter())
# the internal newline is necessary here, not sure why
lintr::lint("if (x > 1)\n  x <- x[-(i)]", lintr::spaces_left_parentheses_linter())
```

All of these are handled by the refactor in #724
